packages/create-cloudflare/templates/svelte/c3.ts

```diff
@@ -39,39 +45,83 @@ const configure = async (ctx: C3Context) => {
 			return false;
 		},
 	});
-	updateStatus(`Changing adapter in ${blue("svelte.config.js")}`);
+};
 
-	// If using typescript, add the platform interface to the `App` interface
-	if (usesTypescript(ctx)) {
-		transformFile("src/app.d.ts", {
-			visitTSModuleDeclaration(n) {
-				if (n.value.id.name === "App") {
-					const patchAst = parseTs(platformInterface);
-					const body = n.node.body as recast.types.namedTypes.TSModuleBlock;
-					body.body.push(patchAst.program.body[0]);
-				}
-
-				this.traverse(n);
-			},
-		});
-		updateStatus(`Updating global type definitions in ${blue("app.d.ts")}`);
+const updateTypeDefinitions = (ctx: C3Context) => {
+	if (!usesTypescript(ctx)) {
+		return;
 	}
+
+	updateStatus(`Updating global type definitions in ${blue("app.d.ts")}`);
+
+	const b = recast.types.builders;
+
+	transformFile("src/app.d.ts", {
+		visitTSModuleDeclaration(n) {
+			if (n.value.id.name === "App" && n.node.body) {
+				const moduleBlock = n.node
+					.body as recast.types.namedTypes.TSModuleBlock;
+
+				const platformInterface = b.tsInterfaceDeclaration(
+					b.identifier("Platform"),
+					b.tsInterfaceBody([
+						b.tsPropertySignature(
+							b.identifier("env"),
+							b.tsTypeAnnotation(b.tsTypeReference(b.identifier("Env")))
+						),
+						b.tsPropertySignature(
+							b.identifier("cf"),
+							b.tsTypeAnnotation(
+								b.tsTypeReference(b.identifier("CfProperties"))
+							)
+						),
+						b.tsPropertySignature(
+							b.identifier("ctx"),
+							b.tsTypeAnnotation(
+								b.tsTypeReference(b.identifier("ExecutionContext"))
+							)
+						),
+					])
+				);
+
+				moduleBlock.body.unshift(platformInterface);
+			}
+
+			this.traverse(n);
+		},
+	});
 };
 
 const config: TemplateConfig = {
 	configVersion: 1,
 	id: "svelte",
 	displayName: "Svelte",
 	platform: "pages",
+	copyFiles: {
+		variants: {
+			js: { path: "./js" },
+			ts: { path: "./ts" },
+		},
+	},
 	generate,
 	configure,
-	transformPackageJson: async () => ({
-		scripts: {
-			"pages:preview": `${npm} run build && wrangler pages dev ${await compatDateFlag()} .svelte-kit/cloudflare`,
-			"pages:deploy": `${npm} run build && wrangler pages deploy .svelte-kit/cloudflare`,
-		},
-	}),
+	transformPackageJson: async (original: PackageJson, ctx: C3Context) => {
+		let scripts: Record<string, string> = {
+			preview: `${npm} run build && wrangler pages dev ${await compatDateFlag()} .svelte-kit/cloudflare`,
```

Comment:
is the compat date necessary when we have it in the toml file?

(I sure hope not because I did remove it in the Next.js template 😅)

Reply:
That's a great point 🙈

Reply:
PS: it might not be that simple thought 😓 

the compat date might not actually properly work with Pages: https://github.com/cloudflare/workers-sdk/issues/4411

I haven't had a chance I should investigate that issue 😓